### ISSUE TITLE
ITMS-90562: Invalid Bundle - using version `2.2.1`

### ISSUE THREAD

**Issue by DanielMandea** [curityio/ios-idsvr-haapi-sdk-dist#3]
While using versions` 2.1.1`, `2.2.0`, and now `2.2.1` of IdsvrHaapiSdk SDK for the Curity Identity Server our apps started  to be rejected by AppStore with the following message: 

“ Invalid Bundle - One or more dynamic libraries that are referenced by your app are not present in the dylib search path. If your app contains bitcode, bitcode processing may have failed. Because of these errors, this build of your app will not be able to be submitted for review or placed on the App Store. “

"ITMS-90562: Invalid Bundle - One or more dynamic libraries that are referenced by your app are not present in the dylib search path. "

"ITMS-90562: Invalid Bundle - The app submission can not be successfully recompiled from bitcode due to missing symbols during linking. You can try to reproduce and diagnose such issues locally by following the instructions from: https://developer.apple.com/library/archive/technotes/tn2432/_index.html "

While investigating this issue we’ve realized that  IdsvrHaapiSdk is not following the latest specifications released by apple regarding building and distributing `XCFrameworks`.
Below there are some indications related to how this issue can be fixed: 
Considering all of the above, I will try to sum up the required steps to prepare successful builds which can be used to create XCFrameworks and later on distributed without problems:

Using archive builds is a must, release build isn't enough
BUILD_LIBRARY_FOR_DISTRIBUTION must be set to YES
SKIP_INSTALL must be set to NO
GCC_INSTRUMENT_PROGRAM_FLOW_ARCS = NO to turn off GCC instrumentation and remove them from the binary
CLANG_ENABLE_CODE_COVERAGE = NO to turn off code coverage tools from the binary 
More details related to this issue can be found within this article: [XCFramework. XCode 12 and BigSur issues.](https://www.scichart.com/blog/xcframework-xcode-12-and-bigsur-issues/)

**Comment by mtrojanowski** [comment]
Hey @DanielMandea, we've released version 2.2.2 which should be fixing your issue. Let us know should there still be a problem.

**Comment by DanielMandea** [comment]
@mtrojanowski Again it's not working with `2.2.2` ...  have you also tried to bake `HAAPI Kit SDK` inside some demo app and then push that to TestFlight?

**Comment by mtrojanowski** [comment]
Yes, we have a test app that uses the SDK and it is passing the Appstore review. It's odd that you still have problems. Is the issue the same as with version 2.2.1, or do you get some other feedback?

**Comment by drager** [comment]
We also get the same issue with 2.2.2.

**Comment by drager** [comment]
@mtrojanowski Any news about this? We're blocked by this.

**Comment by mtrojanowski** [comment]
Hey @drager, @DanielMandea,

We have checked the issue again with our test app. We have an application that uses version 2.2.2 of the SDK. The app is successfully published to TestFlight. Last week we've submitted it for an AppStore release review and it passed it as well. So it seems that from Apple's point of view, an application that uses version 2.2.2 of our SDK could be safely released to the AppStore.

Are you able to verify if it's our SDK that causes the issue? Not sure if Apple shares such concrete information, with an indication of which dependency causes an issue.

**Comment by drager** [comment]
@mtrojanowski Hey, weird... Is this the example app you uploaded to TestFlight? https://github.com/curityio/ios-haapi-demo-app

Well, it started only to be an issue with the version of our app that got this new SDK. Worked fine with versions before 2.2.x, and works fine with other versions of the our app without the curity sdk. We don't get much information from Apple. I can take a closer look in xcode and see if I can find any information about it.

**Comment by mtrojanowski** [comment]
Yes, this is the demo application that we use on TestFlight.

**Comment by drager** [comment]
Alright, seems like version 2.1.0 works fine and the upload erroring happens with 2.1.1 and above.

This seems to be the changelog for 2.1.1:

## Added

- Changelogs for driver and sdk. [HSI-84]
- The framework can be imported and linked in projects targeting deployment versions below iOS 14 when using SPM and drag-and-drop import. [HSI-82]
- Now provides compile time availability check information for OS version

## Fixed

- Crash when running on devices with iOS version below 14 [HSI-82]

**Comment by mtrojanowski** [comment]
The incorrect configuration was introduced in 2.1.1. but it was fixed in 2.2.2. That is odd that you still get the same error when using version 2.2.2 of the SDK.

**Comment by mtrojanowski** [comment]
Hey @drager, @DanielMandea,

Sorry for the delay. It looks like we finally managed to find the cause of the problem. It turns out the issue appeared only when Bitcode was enabled for the application. Previously, we submitted the app with Bitcode disabled.

We've just released version 2.3.0, and hopefully, this will allow you to pass Apple's review. Let me know should you still have any issues.

**Comment by mtrojanowski** [comment]
@drager @DanielMandea I will close this issue for now. Should you run into any more problems with passing the review feel free to comment here or open a new issue.
